### PR TITLE
Added Schultz Fasit handler

### PR DESCRIPTION
**Pull request by jekuaitk**
Adds Schultz Fasit handler

### CODE REVIEW

File: modules/os2forms_fasit/src/Form/SettingsForm.php [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    $fasitApiVersion = $this->settings->getFasitApiVersion();
+    $form[self::FASIT_API_VERSION] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Fasit API version'),
+      '#required' => TRUE,
+      '#default_value' => !empty($fasitApiVersion) ? $fasitApiVersion : NULL,
+      '#description' => $this->t('Specifies which api version to use. Should probably be v2'),
+    ];
+
+    $certificate = $this->settings->getCertificate();
```
Comment (cableman) [inline modules/os2forms_fasit/src/Form/SettingsForm.php:82]: We should real use the key module for this type of data.

Comment (cableman) [reply modules/os2forms_fasit/src/Form/SettingsForm.php:82]: Which have support for different vault backends.

File: modules/os2forms_fasit/src/Helper/CertificateLocatorHelper.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    $locatorType = $certificateSettings[self::LOCATOR_TYPE];
+    $options = $certificateSettings[$locatorType];
+    $options += [
+      self::LOCATOR_PASSPHRASE => $certificateSettings[self::LOCATOR_PASSPHRASE] ?: '',
+    ];
+
+    if (self::LOCATOR_TYPE_AZURE_KEY_VAULT === $locatorType) {
+      $httpClient = new GuzzleAdapter(new Client());
+      $requestFactory = new RequestFactory();
```
Comment (cableman) [inline modules/os2forms_fasit/src/Helper/CertificateLocatorHelper.php:51]: Would one not inject a Drupal::httpClient using PSR18/17 interfaces and not be locked into guzzle?

File: modules/os2forms_fasit/src/Helper/FasitHelper.php [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+      ],
+      'body' => $doc->saveXML(),
+      'cert' => $certificateOptions,
+    ];
+
+    // Attempt upload.
+    try {
+      $response = $this->client->request('POST', $endpoint, $options);
+    }
+    catch (GuzzleException $e) {
```
Comment (cableman) [inline modules/os2forms_fasit/src/Helper/FasitHelper.php:216]: This again locks the module into using guzzle?

**Review by cableman — CHANGES_REQUESTED** [review]

File: modules/os2forms_fasit/src/Form/SettingsForm.php (hunk 1 above)
Comment (jekuaitk) [reply modules/os2forms_fasit/src/Form/SettingsForm.php:82]: Support for keys will be added when it had been finished! 

Currently, it is till work in progress.

File: modules/os2forms_fasit/src/Helper/CertificateLocatorHelper.php (hunk 2 above)
Comment (jekuaitk) [reply modules/os2forms_fasit/src/Helper/CertificateLocatorHelper.php:51]: PSR18/17 does not have built-in mechanisms for working with certificates. This means we would end up having a hard requirement on some sort of adapter (e.g. Guzzle) in order to end up with a PSR18 compliant client.

File: modules/os2forms_fasit/src/Helper/FasitHelper.php (hunk 3 above)
Comment (jekuaitk) [reply modules/os2forms_fasit/src/Helper/FasitHelper.php:216]: See other comment.

**Review by cableman — APPROVED** [review]

**Comment by stankut** [comment]
i was asked to review the code.
good job, well written, i was only able to one potential edge case. If that is though thought, i have no further questions.

File: modules/os2forms_fasit/src/Helper/FasitHelper.php [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+      }
+    }
+
+    [$certificateOptions, $tempCertFilename] = $this->getCertificateOptionsAndTempCertFilename();
+
+    $options = [
+      'headers' => [
+        'Content-Type' => 'application/xml',
+      ],
+      'body' => $doc->saveXML(),
```
[the remark this answers is not in the document]
Comment (jekuaitk) [reply modules/os2forms_fasit/src/Helper/FasitHelper.php]: Added extra check to handle body being FALSE.

**Pull request merged**